src/backend/TrafficCourts/Citizen.Service/Controllers/DisputesController.cs

```diff
@@ -244,47 +358,19 @@ public async Task<IActionResult> UpdateDisputeContactInfoAsync([Required] string
     {
         try
         {
-            // Attempt to decode the hashed noticeOfDisputeGuid
-            var hex = _hashids.DecodeHex(guidHash);
-            if (hex == String.Empty || hex.Length != 32 || !Guid.TryParse(hex, out Guid noticeOfDisputeGuid))
+            if (!_hashids.TryDecodeGuid(guidHash, out Guid noticeOfDisputeGuid))
             {
                 return BadRequest("Invalid guidHash");
             }
 
-            // Verify Dispute exists and whose status is one of [NEW, VALIDATED, or PROCESSING] and there is no corresponding JJDispute
-            string[] validStatuses = { DisputeStatus.NEW.ToString(), DisputeStatus.VALIDATED.ToString(), DisputeStatus.PROCESSING.ToString() };
-            SearchDisputeRequest searchRequest = new() { NoticeOfDisputeGuid = noticeOfDisputeGuid };
-            Response<SearchDisputeResponse> response = await _bus.Request<SearchDisputeRequest, SearchDisputeResponse>(searchRequest, cancellationToken);
-            if (response.Message?.NoticeOfDisputeGuid is null)
-            {
-                return NotFound("Dispute not found");
-            }
-            else if (response.Message?.DisputeStatus is not null && !validStatuses.Contains(response.Message.DisputeStatus))
-            {
-                return BadRequest($"Dispute has a status of {response.Message.DisputeStatus}. Expecting one of NEW, VALIDATED, or PROCESSING.");
-            }
-            else if (response.Message?.JJDisputeStatus is not null)
-            {
-                return BadRequest($"JJDispute has status of {response.Message?.JJDisputeStatus}. Must be blank.");
+            var actionResult = await CheckDisputeStatus(noticeOfDisputeGuid, cancellationToken);

```

Comment:
This is a code smell, one controller action calling another controller action. This is one of the reasons I prefer small controllers with the logic delegated to various services.